internal/fs/vss_windows.go

```diff
@@ -0,0 +1,977 @@
+	VSS_E_SNAPSHOT_NOT_IN_SET                 HRESULT = 0x8004232B
+	VSS_E_MAXIMUM_NUMBER_OF_VOLUMES_REACHED   HRESULT = 0x80042312
+	VSS_E_MAXIMUM_NUMBER_OF_SNAPSHOTS_REACHED HRESULT = 0x80042317
+	VSS_E_NESTED_VOLUME_LIMIT                 HRESULT = 0x8004232C
+	VSS_E_OBJECT_NOT_FOUND                    HRESULT = 0x80042308
+	VSS_E_PROVIDER_NOT_REGISTERED             HRESULT = 0x80042304
+	VSS_E_PROVIDER_VETO                       HRESULT = 0x80042306
+	VSS_E_VOLUME_NOT_SUPPORTED                HRESULT = 0x8004230C
+	VSS_E_VOLUME_NOT_SUPPORTED_BY_PROVIDER    HRESULT = 0x8004230E
+	VSS_E_UNEXPECTED                          HRESULT = 0x80042302
+	VSS_E_UNEXPECTED_PROVIDER_ERROR           HRESULT = 0x8004230F
+	VSS_E_UNSELECTED_VOLUME                   HRESULT = 0x8004232A
+	VSS_E_CANNOT_REVERT_DISKID                HRESULT = 0x800423FE
+	VSS_E_INVALID_XML_DOCUMENT                HRESULT = 0x80042311
+	VSS_E_OBJECT_ALREADY_EXISTS               HRESULT = 0x8004230D
+	FSRVP_E_SHADOW_COPY_SET_IN_PROGRESS       HRESULT = 0x80042316
+)
+
+// hresultToString maps a HRESULT value to a human readable string.
+var hresultToString = map[HRESULT]string{
+	S_OK:                                    "S_OK",
+	E_ACCESSDENIED:                          "E_ACCESSDENIED",
+	E_OUTOFMEMORY:                           "E_OUTOFMEMORY",
+	E_INVALIDARG:                            "E_INVALIDARG",
+	VSS_E_BAD_STATE:                         "VSS_E_BAD_STATE",
+	VSS_E_LEGACY_PROVIDER:                   "VSS_E_LEGACY_PROVIDER",
+	VSS_E_RESYNC_IN_PROGRESS:                "VSS_E_RESYNC_IN_PROGRESS",
+	VSS_E_SNAPSHOT_NOT_IN_SET:               "VSS_E_SNAPSHOT_NOT_IN_SET",
+	VSS_E_MAXIMUM_NUMBER_OF_VOLUMES_REACHED: "VSS_E_MAXIMUM_NUMBER_OF_VOLUMES_REACHED",
+	VSS_E_MAXIMUM_NUMBER_OF_SNAPSHOTS_REACHED: "VSS_E_MAXIMUM_NUMBER_OF_SNAPSHOTS_REACHED",
+	VSS_E_NESTED_VOLUME_LIMIT:                 "VSS_E_NESTED_VOLUME_LIMIT",
+	VSS_E_OBJECT_NOT_FOUND:                    "VSS_E_OBJECT_NOT_FOUND",
+	VSS_E_PROVIDER_NOT_REGISTERED:             "VSS_E_PROVIDER_NOT_REGISTERED",
+	VSS_E_PROVIDER_VETO:                       "VSS_E_PROVIDER_VETO",
+	VSS_E_VOLUME_NOT_SUPPORTED:                "VSS_E_VOLUME_NOT_SUPPORTED",
+	VSS_E_VOLUME_NOT_SUPPORTED_BY_PROVIDER:    "VSS_E_VOLUME_NOT_SUPPORTED_BY_PROVIDER",
+	VSS_E_UNEXPECTED:                          "VSS_E_UNEXPECTED",
+	VSS_E_UNEXPECTED_PROVIDER_ERROR:           "VSS_E_UNEXPECTED_PROVIDER_ERROR",
+	VSS_E_UNSELECTED_VOLUME:                   "VSS_E_UNSELECTED_VOLUME",
+	VSS_E_CANNOT_REVERT_DISKID:                "VSS_E_CANNOT_REVERT_DISKID",
+	VSS_E_INVALID_XML_DOCUMENT:                "VSS_E_INVALID_XML_DOCUMENT",
+	VSS_E_OBJECT_ALREADY_EXISTS:               "VSS_E_OBJECT_ALREADY_EXISTS",
+	FSRVP_E_SHADOW_COPY_SET_IN_PROGRESS:       "FSRVP_E_SHADOW_COPY_SET_IN_PROGRESS",
+}
+
+// Str converts a HRESULT to a human readable string.
+func (h HRESULT) Str() string {
+	if i, ok := hresultToString[h]; ok {
+		return i
+	} else {
+		return "UNKNOWN"
```

Comment:
complaint by golint: " if block ends with a return statement, so drop this else and outdent its block"

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/f70a1dfd172d54843e27deae292f3557d11ade84